frontend/tests/e2e-tests.sh

```
@@ -96,78 +94,78 @@ s.close()
-			echo "The port $PORT is already in use!"
-			echo "Please stop the running process using the port or change the mailer port and try again."
-			exit 1
-		fi
-	elif [[ " ${SCRIPT_SHORT_ARGS[@]} " =~ " -m " ]]; then
-		echo "No mailer service is running on port $PORT!"
-		echo "Please start a mailer service on port $PORT or change the mailer port using --mailer=PORT/PORT and try again."
-		echo "You can also use the isolated test mailer service by removing the -m option."
-		exit 1
-	fi
+  if python3 -c "import socket;exit(0 if 0 == socket.socket(socket.AF_INET, socket.SOCK_STREAM).connect_ex(('localhost',$PORT)) else 1)"; then
+    if [[ ! " ${SCRIPT_SHORT_ARGS[@]} " =~ " -m " ]]; then
```

Shellcheck (0.10.0)

[error] Arrays implicitly concatenate in [[ ]]. Use a loop (or explicit * instead of @).

(SC2199)